projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -1087,6 +1087,22 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
     @Output()
     public onGroupingDone = new EventEmitter<ISortingExpression[]>();
 
+    /**
+     * Emitted when a new `IgxColumnComponent` is grouped or ungrouped.
+     * Returns the `ISortingExpression` related to the grouping or ungrouping operation.
+     * ```typescript
+     * groupingChanged(event: any){
+     *     const grouping = event;
+     * }
+     * ```
+     * ```html
+     * <igx-grid #grid [data]="localData" (onGroupingChanged)="groupingChanged($event)" [autoGenerate]="true"></igx-grid>
+     * ```
+	 * @memberof IgxGridComponent
+     */
+    @Output()
+    public onGroupingChanged = new EventEmitter<ISortingExpression[]>();
```

Comment:
I thought there's going to be more than just the sorting expressions in the grouping event arguments. Shouldn't there be column components?

Reply:
@PavlovVasil See https://github.com/IgniteUI/igniteui-angular/issues/2447#issuecomment-419945534 and https://github.com/IgniteUI/igniteui-angular/issues/2447#issuecomment-419947496